evap/results/tests/test_tools.py

```diff
@@ -151,10 +157,17 @@ def test_average_deviation(self):
         mommy.make(RatingAnswerCounter, question=question_likert, contribution=contribution1, answer=5, count=4)
         mommy.make(RatingAnswerCounter, question=question_likert, contribution=general_contribution, answer=5, count=3)
 
-        __, deviation = calculate_average_grades_and_deviation(course)
+        # contribution1: 0.4 * (0.5, 0, 0.5, 0, 0) + 0.6 * (0, 0, 0.5, 0, 0.5) = (0.2, 0, 0.5, 0, 0.3)
+        # contribution2: (0, 0.5, 0, 0.5, 0)
+        # contributions: (0.1, 0.25, 0.25, 0.25, 0.15)
+
+        # course_non_grade: (0, 0, 0, 0, 1)
 
-        total_likert_dev = settings.CONTRIBUTION_PERCENTAGE * 1 + (1 - settings.CONTRIBUTION_PERCENTAGE) * 0
-        total_grade_dev = 1
-        total_dev = settings.GRADE_PERCENTAGE * total_grade_dev + (1 - settings.GRADE_PERCENTAGE) * total_likert_dev
+        # total: 0.375 * (0.1, 0.25, 0.25, 0.25, 0.15) + 0.625 * (0, 0, 0, 0, 1) = (0.0375, 0.09375, 0.09375, 0.09375, 0.68125)
 
-        self.assertAlmostEqual(deviation, total_dev)
+        __, distribution = calculate_average_grade_and_distribution(course)
+        self.assertAlmostEqual(distribution[1], 0.0375)
+        self.assertAlmostEqual(distribution[2], 0.09375)
+        self.assertAlmostEqual(distribution[3], 0.09375)
+        self.assertAlmostEqual(distribution[4], 0.09375)
+        self.assertAlmostEqual(distribution[5], 0.68125)
```

Comment:
okay, but i'll add an additional test, so that we also see whether not having all three components behaves as expected.

Reply:
even better, thanks.

Reply:
...the ultimate test whether we got this right would be to add up the distribution percentages multiplied with their index, and compare that with the average grade coming from the calculation function, wouldn't it? actually that would be kinda important to test, since we wanted the distribution to be accurate...

Reply:
check